File: client/src/models/OpenHoursViewModel.ts
import { z } from "zod";

const OpenHoursViewModel = z.object({
    mondayOpenTime: z.string().nullable(),
    mondayCloseTime: z.string().nullable(),
    tuesdayOpenTime: z.string().nullable(),
    tuesdayCloseTime: z.string().nullable(),
    wednesdayOpenTime: z.string().nullable(),
    wednesdayCloseTime: z.string().nullable(),
    thursdayOpenTime: z.string().nullable(),
    thursdayCloseTime: z.string().nullable(),
    fridayOpenTime: z.string().nullable(),
    fridayCloseTime: z.string().nullable(),
    saturdayOpenTime: z.string().nullable(),
    saturdayCloseTime: z.string().nullable(),
    sundayOpenTime: z.string().nullable(),
    sundayCloseTime: z.string().nullable(),
})

export default OpenHoursViewModel;
export type OpenHoursViewModelType = z.infer<typeof OpenHoursViewModel>;

File: Server/src/Server.Application/ViewModels/OpenHoursViewModel.cs
namespace Server.Application.ViewModels;

public class OpenHoursViewModel
{
    public TimeSpan? MondayOpenTime { get; set; }
    public TimeSpan? MondayCloseTime { get; set; }
    public TimeSpan? TuesdayOpenTime { get; set; }
    public TimeSpan? TuesdayCloseTime { get; set; }
    public TimeSpan? WednesdayOpenTime { get; set; }
    public TimeSpan? WednesdayCloseTime { get; set; }
    public TimeSpan? ThursdayOpenTime { get; set; }
    public TimeSpan? ThursdayCloseTime { get; set; }
    public TimeSpan? FridayOpenTime { get; set; }
    public TimeSpan? FridayCloseTime { get; set; }
    public TimeSpan? SaturdayOpenTime { get; set; }
    public TimeSpan? SaturdayCloseTime { get; set; }
    public TimeSpan? SundayOpenTime { get; set; }
    public TimeSpan? SundayCloseTime { get; set; }
}
